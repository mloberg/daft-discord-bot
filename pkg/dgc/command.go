package dgc

import "github.com/bwmarrin/discordgo"

// Command represents a Discord slash command
type Command struct {
	Name        string
	Description string
	Options     []*discordgo.ApplicationCommandOption

	Run func(s *discordgo.Session, i *discordgo.InteractionCreate) error
}
