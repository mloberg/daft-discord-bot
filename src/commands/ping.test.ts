import { Client, Message, TextChannel } from 'discord.js';

import command from './ping';

const mocks = {
    react: jest.fn(),
};

jest.mock('discord.js', () => {
    return {
        Message: jest.fn().mockImplementation(() => {
            return {
                react: mocks.react,
            };
        }),
    };
});

describe('_ping', () => {
    let message: Message;

    beforeEach(() => {
        mocks.react.mockClear();

        const client = {} as Client;
        const channel = {} as TextChannel;
        message = new Message(client, {}, channel);
    });

    it('is a command', () => {
        expect(command.name).toBe('ping');
        expect(command).toMatchSnapshot();
    });

    it('responds', async () => {
        await command.run(message, { _: [], $0: 'ping' });

        expect(mocks.react).toBeCalledWith('👍');
    });
});
