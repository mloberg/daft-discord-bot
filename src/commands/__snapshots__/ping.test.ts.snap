// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`_ping is a command 1`] = `
Object {
  "description": "Pong",
  "name": "ping",
  "run": [Function],
}
`;
